py3status/modules/speedtest_cli.py

```diff
@@ -0,0 +1,294 @@
+
+
+The module will be triggered on clic only. Not at start.
+
+Note that all placeholders have a version prefixed by `previous_`, eg `previous_download`.
+This can be usefull to compare two runs.
+
+Requires:
+    speedtest-cli: Command line interface for testing internet bandwidth using speedtest.net
+
+Examples:
+```
+# colored based on thresholds
+speedtest_cli {
+    thresholds = {
+        'download': [
+            (0, 'bad'), (1024, 'degraded'), (1024 * 1024, 'good')
+        ],
+        'ping': [
+            (200, 'bad'), (150, 'orange'), (100, 'degraded'), (10, 'good')
+        ],
+        'quality': [
+            ("ok", "good"),
+            ("bad", "degraded"),
+            ("faster", "good"),
+            ("slower", "degraded"),
+            ("faster", "good"),
+        ],
+        'upload': [
+            (0, 'bad'), (1024, 'degraded'), (1024 * 1024, 'good')
+        ],
+    }
+    format = '[\?color=ping {ping} ms] [\?color=upload ↑ {upload} {upload_unit}]'
+    format += ' [\?color=download ↓ {download} {download_unit}]'
+}
+
+# colored based on comparation between two last runs
+speedtest_cli {
+    format = ' [[\?if=download_raw<previous_download_raw&color=degraded ↓]|[\?color=ok ↑]]'
+}
+
+# compare only download
+speedtest_cli {
+    format = '[[\?if=quality=slower&color=download ↓ {download} {download_unit}]'
+    format += '[\?if=quality=faster&color=download ↑ {download} {download_unit}]'
+    format += '[\?if=quality=ok&color=download -> {download} {download_unit}]'
+    format += '[\?if=quality=bad&color=download x {download} {download_unit}]]'
+    format += '|'
+}
+```
+
+@author Cyril Levis (@cyrinux)
+
+SAMPLE OUTPUT
+{'full_text': u'\u25cf 208.18 ms \u2191 0.247 MB/s \u2193 0.453 MB/s'}
+"""
+
+from json import loads
+
+STRING_NOT_INSTALLED = "not installed"
+MISSING_PLACEHOLDER = "{download} or {upload} placeholder required"
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    button_refresh = 1
+    button_share = None
+    format = (
+        u"\u25cf [{ping} ms] [↑ {upload} {upload_unit}] [↓ {download} {download_unit}]"
+    )
+    server_id = None
+    si_units = False
+    sleep_timeout = 5
+    thresholds = {
+        "download": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+        "ping": [(200, "bad"), (150, "orange"), (100, "degraded"), (10, "good")],
+        "quality": [
+            ("ok", "good"),
+            ("bad", "degraded"),
+            ("faster", "good"),
+            ("slower", "degraded"),
+            ("faster", "good"),
+        ],
+        "upload": [(0, "bad"), (1024, "degraded"), (1024 * 1024, "good")],
+    }
+    timeout = 10
+    unit_bitrate = "MB/s"
+    unit_size = "MB"
+
+    def post_config_hook(self):
+        if not self.py3.check_commands("speedtest-cli"):
+            raise Exception(STRING_NOT_INSTALLED)
+
+        self.first_run = True
+        self.placeholders = list(set(self.py3.get_placeholders_list(self.format)))
```

Comment:
I think it already does this. You can test this in terminal.
```diff
diff --git a/py3status/modules/static_string.py b/py3status/modules/static_string.py
index dbcec8c6..c9d6f8bc 100644
--- a/py3status/modules/static_string.py
+++ b/py3status/modules/static_string.py
@@ -17,8 +17,13 @@ class Py3status:
     """
     # available configuration parameters
     format = 'Hello, world!'
+    format += '{a} {d}'
+    format += '{b} {c}'
+    format += '{c} {b}'
+    format += '{d} {a}'

     def static_string(self):
+        self.py3.log(self.py3.get_placeholders_list(self.format))
         return {
             'cached_until': self.py3.CACHE_FOREVER,
             'full_text': self.py3.safe_format(self.format),
```

Reply:
Find this in another module, not test but i feel your are true.